homeassistant/helpers/script.py

```diff
@@ -455,10 +457,9 @@ async def _async_wait_template_step(self):
         @callback
         def async_script_wait(entity_id, from_s, to_s):
             """Handle script after template condition is true."""
-            self._variables["wait"] = {
-                "remaining": to_context.remaining if to_context else timeout,
-                "completed": True,
-            }
+            wait_var = self._variables["wait"]
```

Comment:
Sneaky, updating the dict that's also referenced in the trace 😉

Reply:
Yeah, is it OK or do you think it's confusing and/or will cause problems?

Reply:
Same as https://github.com/home-assistant/core/pull/48100#discussion_r598661151